README.md

```diff
@@ -109,135 +49,220 @@ You should use the following process as much as possible:
 
-**Exercise:** Look at this requirement in Wave 1: "For a given user, calculate their total expenditure for all trips". Spend some time writing pseudocode for this.
+### Reading Code
 
-### Wave 1
+The existing code contains 4 classes:
+- `Passenger`
+- `Trip`
+- `CsvRecord`
+- `TripDispatcher`
+
+**Before you go any further, think about how these classes might be related.** Draw a diagram, if that will help. Your guess doesn't have to be correct at this stage - making a prediction and then checking it is an important part of the learning process.
+
+Now, start reading through the code. There are two equally valid ways to approach this:
+
+- Top-down: start with the big pieces.
+    - What classes are there, and how do they interact?
+    - How does it look from the outside?
+    - How are these classes tested?
+- Bottom-up: start with the details. Pick an interesting method (say `Passenger.load_all`) and learn everything about it.
+    - What arguments does it take?
+    - How does it work? If there's more than one step, what are they? Why are they in that order?
+    - What other methods does call? 
+    - What does it return?
+    - What code calls it, and how does it use the returned value?
+    - How is it tested?
+
+### Playing Around
+
+One of the most effective way to learn a new codebase is to use it. Since this code is a library not a program, there's nothing obvious to run from the command line, so we'll have to be a little clever. We can use `pry` to load the code, and then play with it in the interpreter:
+
+```
+$ pry -r ./lib/trip_dispatcher.rb
+[1] pry(main)> td = RideShare::TripDispatcher.new
+=> #<RideShare::TripDispatcher:0x3fe91f52950c>
+[2] pry(main)> td.trips.length
+=> 600
+```
+
+### Comprehension Questions
+
+Before you start writing code, you and your partner should work together, using the above techniques to answer the following questions. If you have questions or run into something that you can't figure out, consult with another group.
+
+#### Classes and Relationships
+
+- What _inheritance_ relations exist between classes?
+- What _composition_ relations exist between classes?
+- Do these relations match your prediction from earlier?
+- Draw a class diagram that contains all of the above relations.
+
+#### Code Details
+
+- Why doesn't `Passenger` or `Trip` need an `attr_reader` for `id`?
+- Why does the version of `from_csv` in `CsvRecord` raise a `NotImplementedError`? What does this mean? Why don't we hit that when we run the code?
+- Why is `from_csv` a private method?
+- How does `CsvRecord.load_all` know what CSV file to open? 
+- When you call `Passenger.load_all`, what happens? What methods are called in what order?
+
+#### Using the Library
+
+Using the pry session we started above, how would you...
+
+- Print the rating for the first trip
+- Print the name of the passenger for trip 7
+- Print the ID of the trip that cost the most money
+- Print the name the passenger who has spent the most money
```

Comment:
Hm, the two bullet points of "... the most money" prompts seem pretty difficult to do as a warm-up exercise, maybe? Would it make sense to only have the first two prompts? My concern is that I think pairs that are struggling wouldn't be able to do this in the first day and a half.